drivers/cadence/uart/cdns_console.S

```diff
@@ -0,0 +1,127 @@
+/*
```

Comment:
There is some *rXpY* revision information (see: https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/serial/cdns,uart.txt), which I'm not sure whether it should go into the filename as the differences might be small and not requiring a whole new driver for each rev.
I'm not aware of any other SoC using this IP. I can move it to plat/xilinx/drivers or something like that.